### DISCUSSION TITLE
async subquery error in relationship

### DISCUSSION THREAD

**Discussion by wu-clan** [sqlalchemy/sqlalchemy#7835] [Usage Questions] [answered]
**for many to many**
in table Role:
`users = relationship('User', backref='roles', secondary='user_role')`
select():
`user = await user_crud.get_user_by_id(db, user_id)`
i use `user.roles`:
sqlalchemy.exc.StatementError: (sqlalchemy.exc.MissingGreenlet) greenlet_spawn has not been called; can't call await_() here. Was IO attempted in an unexpected place?
[SQL: SELECT `role`.id AS role_id, `role`.name AS role_name, `role`.description AS role_description 
FROM `role`, user_role 
WHERE %s = user_role.user_id AND `role`.id = user_role.role_id]
[parameters: [{'%(2182874338736 param)s': 2}]]
(Background on this error at: https://sqlalche.me/e/14/xd2s)

**Comment by CaselIT** [comment, marked as answer]
Hi,

Following the link on the error you get to https://docs.sqlalchemy.org/en/14/errors.html#error-xd2s then to https://docs.sqlalchemy.org/en/14/orm/extensions/asyncio.html#asyncio-orm-avoid-lazyloads that illustrates how to eager load data when using relationships with asyncio. Hope it helps
